pkg/csi/service/wcpguest/controller.go

```diff
@@ -332,12 +335,30 @@ func (c *controller) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequ
 				c.supervisorNamespace, err)
 			log.Error(msg)
 			eventList, err := c.supervisorClient.CoreV1().Events(c.supervisorNamespace).List(ctx,
-				metav1.ListOptions{FieldSelector: "involvedObject.name=" + pvc.Name})
+				metav1.ListOptions{
+					FieldSelector:        "involvedObject.name=" + pvc.Name,
+					ResourceVersion:      pvc.ResourceVersion,
+					ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan,
+				})
 			if err != nil {
 				log.Errorf("Unable to fetch events for pvc %q/%q from supervisor cluster with err: %+v",
 					c.supervisorNamespace, pvc.Name, err)
 				return nil, csifault.CSIInternalFault, status.Errorf(codes.Internal, msg)
 			}
+
+			var failureMessage string
+			for _, svcPvcEvent := range eventList.Items {
+				// todo: should we handle this check differently than using strings.Contains?
+				if strings.Contains(svcPvcEvent.Reason, pvcFailureReason) {
```

Comment:
Instead of performing the string Contains check, if we do the event Type checking, we can directly get the event.Message.

Reply:
But how do we know there couldn't be any other `Warning` events?

Reply:
Checking the different types of volume related events (k8s.io/kubernetes/pkg/controller/volume/events/event.go), I do see the following events other than `ProvisioningFailed` -  `ProvisioningCleanupFailed`, `VolumeFailedDelete`, `VolumeMismatch`, `FailedBinding`. 

Ok, I'll change the check to `event.Type == Warning` only

Reply:
Quick check on go playground and it works correctly -> https://go.dev/play/p/OT1723wD1IF